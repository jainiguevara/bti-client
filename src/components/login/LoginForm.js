import React from 'react'
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { connect } from 'react-redux'
import compose from 'lodash/fp/compose'
import { Button, FormControl, TextField, Input, InputLabel, IconButton, InputAdornment } from '@material-ui/core'
import { Visibility, VisibilityOff } from '@material-ui/icons'
import { withStyles } from '@material-ui/core/styles';

import history from './../../history'
import configureStore from './../../store/configureStore';
import { submitLogin } from './../../actions/user'
import { isLoggedIn } from './../../selectors/user'

// const store = configureStore()

const styles = theme => ({
  margin: {
    margin: theme.spacing.unit,
  },
  textField: {
    width: 200
  },
})

class LoginForm extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: ''
    }
  }

  handleSubmit = e => {
    // add validation handling here
    submitLogin({
      email: this.state.email,
      password: this.state.password
    }, this.props.dispatch).then(() => {
      window.location = '/'
    }).catch(e => {
      console.log(e);
    })
  }

  handleEmailChange = e => {
    const email = e.target.value
    this.setState({
      email
    });
  };

  handlePasswordChange = e => {
    const password = e.target.value
    this.setState({
      password,
    });
  }

  handleMouseDownPassword = e => {
    e.preventDefault();
  };

  handleClickShowPassword = () => {
    this.setState(state => ({ showPassword: !state.showPassword }));
  };

  render() {
    const { classes } = this.props
    return (
      <div className={classNames(classes.appFrame)}>
          <TextField
            id="email"
            label="Email"
            className={classNames(classes.margin, classes.textField)}
            value={this.state.email}
            onChange={this.handleEmailChange}
            margin="normal"
            autoFocus
          /> 
          <FormControl className={classNames(classes.margin, classes.textField)}>
            <InputLabel htmlFor="adornment-password">Password</InputLabel>
            <Input
              id="adornment-password"
              type={this.state.showPassword ? 'text' : 'password'}
              value={this.state.password}
              onChange={this.handlePasswordChange}
              endAdornment={
                <InputAdornment position="end">
                  <IconButton
                    aria-label="Toggle password visibility"
                    onClick={this.handleClickShowPassword}
                    onMouseDown={this.handleMouseDownPassword}
                  >
                    {this.state.showPassword ? <VisibilityOff /> : <Visibility />}
                  </IconButton>
                </InputAdornment>
              }
            />
          </FormControl>
          <Button 
            variant="contained" 
            color="secondary" 
            className={classes.button}
            onClick={this.handleSubmit}
          >
            Login
          </Button>
      </div>
    )
  }
}

LoginForm.propTypes = {
  classes: PropTypes.object.isRequired,
};

const mapStateToProps = state => {
  return {
    user : state.user
  }
}

// const mapDispatchToProps = dispatch => {
//   return {
    
//   }
// }

export default compose(
  withStyles(styles),
  connect(mapStateToProps)
)(LoginForm)