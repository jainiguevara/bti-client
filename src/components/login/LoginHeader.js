import React from 'react'
import { Typography } from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";

const styles = theme => ({
  root: {
    margin: theme.spacing.unit,
  }
})

const LoginHeader = (props) => {
  const { classes } = props
  return (
  <div className={classes.root}>
    <Typography variant="title">BTI COURIER EXPRESS INC. - API</Typography>
  </div>  
  )
}

export default withStyles(styles)(LoginHeader)