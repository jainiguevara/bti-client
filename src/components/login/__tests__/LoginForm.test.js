import React from 'react'
import renderer from 'react-test-renderer'
import LoginForm from '../LoginForm'
import LoginHeader from '../LoginHeader'


it('should render the LoginForm', () => {
  const component =
    renderer.create(
      <LoginHeader />
    )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})