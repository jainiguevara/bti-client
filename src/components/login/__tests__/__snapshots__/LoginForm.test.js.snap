// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the LoginForm 1`] = `
<div
  className="LoginHeader-root-1"
>
  <h2
    className="MuiTypography-root-2 MuiTypography-title-8"
  >
    BTI COURIER EXPRESS INC. - API
  </h2>
</div>
`;
